src/Microsoft.Framework.Configuration/ConfigurationBuilder.cs

```diff
@@ -24,7 +25,7 @@ public ConfigurationBuilder(string basePath, params IConfigurationSource[] sourc
                 }
             }
 
-            BasePath = basePath;
+            _properties["BasePath"] = string.Empty;
```

Comment:
We shouldn't need to do this, i.e. ConfigurationBuilder shouldn't need to know about any specific properties.